### PR TITLE
Fallback action-runner to autoupdate when it's unable to start

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Clean the `_work` directory between runner's launches. Fallback to auto-update actions runner if it fails to start. Make the `init-network.sh` sourceable and executable.

### TODO

- [ ] Add the tailscale hostname to the actions-hooks/pre-run.sh
- [ ] Make a file with tailscale hostname in setup_tailscale function to source
- [ ] On quick-finish jobs like `Cancel`, it finishes so quickly that the ATTEMPT is wrongly increased. Doesn't harm, though it should be improved.

### Note
On April 9th, the whole infrastructure failed because GitHub just passed responsibility for updating the runner software to users without any way to identify how soon the hard deprecation would be effective.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit cbadc00735ef6a1a2ff7af51434a2b94c016dea8 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63195/cbadc00735ef6a1a2ff7af51434a2b94c016dea8/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/worker/init_runner.sh [hunk 1]
```diff
@@ -300,23 +300,46 @@ list_children () {
...
-        sudo -u ubuntu ./config.sh --url $RUNNER_URL --token "$(get_runner_token)" \
-          --ephemeral --disableupdate --unattended \
-          --runnergroup Default --labels "$LABELS" --work _work --name "$INSTANCE_ID"
+        token_args=(--token "$(get_runner_token)")
+        config_args=(
+            "${token_args[@]}" --url "$RUNNER_URL"
+            --ephemeral --unattended --replace --runnergroup Default
+            --labels "$LABELS" --work _work --name "$INSTANCE_ID"
+        )
+        if (( ATTEMPT > 1 )); then
```
Comment (Felixoid) [inline tests/ci/worker/init_runner.sh:334]: I need to handle it correctly. Unfortunately, the `run.sh` does not provide information on whether the runner has finished successfully or failed. See https://github.com/actions/runner/issues/3266

So, there's the next plan:

- replace run.sh by `./bin/Runner.Listener` and write the last status into some temp file
- Read it and increase the ATTEMPT if last time it failed

**Comment by Felixoid** [comment]
The staging is deployed from 5f1a72d6df5405ac7991047117731cddadeca5ba

**Comment by Felixoid** [comment]
The production is deployed from baffd31423f5d64e47ef53a9232406fa91ba4060 on May 14, 2024, 12:35:51 CEST

**Pull request merged**